public/app/AppWrapper.tsx

```diff
@@ -144,3 +148,111 @@ export class AppWrapper extends Component<AppWrapperProps, AppWrapperState> {
     );
   }
 }
+
+function RouterTree(props: { routes?: JSX.Element | false }) {
+  return (
+    <Router history={locationService.getHistory()}>
+      <LocationServiceProvider service={locationService}>
+        <CompatRouter>
+          <AppChrome>
+            <AngularRoot />
+            <AppNotificationList />
+            <Stack gap={0} grow={1} direction="column">
+              {pageBanners.map((Banner, index) => (
+                <Banner key={index.toString()} />
+              ))}
+              {props.routes}
+            </Stack>
+            {bodyRenderHooks.map((Hook, index) => (
+              <Hook key={index.toString()} />
+            ))}
+          </AppChrome>
+        </CompatRouter>
+      </LocationServiceProvider>
+    </Router>
+  );
+}
+
+/**
+ * Renders both the main app tree and a secondary sidecar app tree to show 2 apps at the same time in a resizable split
+ * view.
+ * @param props
+ * @constructor
+ */
+function ExperimentalSplitPaneTree(props: { routes?: JSX.Element | false }) {
```

Comment:
Maybe this should live in another file?

Reply:
So I pushed it out but also had to move the main baseline wrapper to prevent cyclic imports (probably would work fine but still don't think it's a good idea to have those if avoidable)